tests/test_cli_compile.py

```diff
@@ -1215,6 +1216,97 @@ def test_generate_hashes_with_line_style_annotations(runner):
     )
 
 
+PYPI_SIMPLE_SMALL_FAKE_A = {
+    "files": [
+        {
+            "filename": "small_fake_a-0.1-py2.py3-none-any.whl",
+            "hashes": {
+                "sha256": "33e1acdca3b9162e002cedb0e58b350d731d1ed3f53a6b22e0a628bca7c7c6ed"
+            },
+            "requires-python": "",
+            "url": "https://files.pythonhosted.org/packages/08/e3/"
+            "ad05e1371eb99f257ca00f791b755deb22e752393eb8e75bc01d651715b02ea9/"
+            "small_fake_a-0.1-py2.py3-none-any.whl",
+            "yanked": False,
+        },
+    ]
+}
+
+PYPI_PROJECT_SMALL_FAKE_A = {
+    "releases": {
+        "0.1": [
+            {
+                "packagetype": "bdist_wheel",
+                "digests": {
+                    "sha256": "33e1acdca3b9162e002cedb0e58b350d731d1ed3f53a6b22e0a628bca7c7c6ed",
+                },
+                "url": (
+                    "https://files.pythonhosted.org/packages/08/e3/"
+                    "ad05e1371eb99f257ca00f791b755deb22e752393eb8e75bc01d651715b02ea9/"
+                    "small_fake_a-0.1-py2.py3-none-any.whl"
+                ),
+            },
+        ]
+    }
+}
+
+
+def test_generate_hashes_with_mixed_sources(runner):
+    with open("requirements.in", "w") as fp:
+        fp.write("small_fake-a==0.1")
+
+    with (
+        MockRequests()
+        .handle(
+            "https://pypi.org/simple/small-fake-a/",
+            MockRequests.response_json(
+                PYPI_SIMPLE_SMALL_FAKE_A,
+                content_type="application/vnd.pypi.simple.v1+json",
+            ),
+        )
+        .handle(
+            "https://pypi.org/pypi/small-fake-a/json",
+            MockRequests.response_json(PYPI_PROJECT_SMALL_FAKE_A),
+        )
+        .handle(
+            "https://pypi.org/pypi/small_fake-a/json",
+            MockRequests.response_json(PYPI_PROJECT_SMALL_FAKE_A),
+        )
+        .handle(
```

Comment:
Mocks should not be used in integration tests. I'll suggest later a solution on how to avoid mocks. Thanks for the tests anyway!

Reply:
I've added a test without mocks, though it's much heavier in terms of network bandwidth and time.

Reply:
I've simplified the test, see https://github.com/neykov/pip-tools/pull/1